ArchiSteamFarm/IPC/Integration/ApiAuthenticationMiddleware.cs

```diff
@@ -88,7 +89,45 @@ internal sealed class ApiAuthenticationMiddleware {
 		await context.Response.WriteJsonAsync(new GenericResponse<StatusCodeResponse>(false, statusCodeResponse), jsonOptions.Value.SerializerSettings).ConfigureAwait(false);
 	}
 
-	private static void ClearFailedAuthorizations(object? state = null) => FailedAuthorizations.Clear();
+	internal static void ClearFailedAuthorizations(object? state = null) => FailedAuthorizations.Clear();
+
+	internal static HashSet<IPAddress> GetCurrentlyBannedIPs() => FailedAuthorizations.Where(static kv => kv.Value >= MaxFailedAuthorizationAttempts).Select(static kv => kv.Key).ToHashSet();
+
+	internal static async Task<bool> UnbanIP(IPAddress ipAddress) {
+		ArgumentNullException.ThrowIfNull(ipAddress);
+
+		if (!FailedAuthorizations.TryGetValue(ipAddress, out byte attempts) || (attempts < MaxFailedAuthorizationAttempts)) {
+			return false;
+		}
+
+		while (true) {
```

Comment:
You overcomplicated this for no reason, just return result of `FailedAuthorizations.TryRemove()`. You don't need my super advanced logic which limits access against DDoS-like situations.